openmetadata-service/src/main/java/org/openmetadata/service/migration/utils/v153/MigrationUtil.java

```diff
@@ -11,4 +16,27 @@ public static void updateEmailTemplates(CollectionDAO collectionDAO) {
     // delete emailTemplates, it will be loaded from initSeedData.
     dao.deleteEmailTemplates();
   }
+
+  public static void updateUserNameToEmailPrefixForLdapAuthProvider(
+      CollectionDAO daoCollection, AuthenticationConfiguration config) {
+    if (config.getProvider().equals(AuthProvider.LDAP)) {
+      LOG.info("Starting migration username -> email prefix");
+      int total = daoCollection.userDAO().listTotalCount();
+      int offset = 0;
+      int limit = 200;
+      while (offset < total) {
+        List<String> userEntities = daoCollection.userDAO().listAfterWithOffset(limit, offset);
+        for (String json : userEntities) {
+          User userEntity = JsonUtils.readValue(json, User.class);
+          String email = userEntity.getEmail();
+          String emailPrefix = email.substring(0, email.indexOf("@"));
+          userEntity.setFullyQualifiedName(emailPrefix);
```

Comment:
here don't put email prefix directly, use as below

```userEntity.setFullyQualifiedName(EntityInterfaceUtil.quote(emailPrefix)); ```

also update the name here without 

``` userEntity.setFullyQualifiedName(emailPrefix); ```

Reply:
please note name won't need quote thing
Also you might need to if we need to resync the nameHash for the user_entity table after this update